packages/statemanager/src/simpleStateManager.ts

```diff
@@ -0,0 +1,198 @@
+ *
+ * This state manager comes with the basic state access logic for
+ * accounts, storage and code (put* and get* methods) as well as a simple
+ * implementation of checkpointing but lacks methods implementations of
+ * state root related logic as well as some other non-core functions.
+ *
+ * Functionality provided is sufficient to be used for simple EVM use
+ * cases and the state manager is used as default there.
+ *
+ * For a more full fledged and MPT-backed state manager implementation
+ * have a look at the `@ethereumjs/statemanager` package.
+ */
+export class SimpleStateManager implements EVMStateManagerInterface {
+  public accountStack: Map<PrefixedHexString, Account | undefined>[] = []
+  public codeStack: Map<PrefixedHexString, Uint8Array>[] = []
+  public storageStack: Map<string, Uint8Array>[] = []
+
+  originalStorageCache: {
+    get(address: Address, key: Uint8Array): Promise<Uint8Array>
+    clear(): void
+  }
+
+  public readonly common?: Common
+
+  constructor(opts: SimpleStateManagerOpts = {}) {
+    this.checkpointSync()
+    this.originalStorageCache = new OriginalStorageCache(this.getContractStorage.bind(this))
+    this.common = opts.common
+  }
+
+  protected topAccountStack() {
+    return this.accountStack[this.accountStack.length - 1]
+  }
+  protected topCodeStack() {
+    return this.codeStack[this.codeStack.length - 1]
+  }
+  protected topStorageStack() {
+    return this.storageStack[this.storageStack.length - 1]
+  }
+
+  // Synchronous version of checkpoint() to allow to call from constructor
+  protected checkpointSync() {
+    const newTopA = new Map(this.topAccountStack())
+    for (const [address, account] of newTopA) {
+      const accountCopy =
+        account !== undefined
+          ? Object.assign(Object.create(Object.getPrototypeOf(account)), account)
+          : undefined
+      newTopA.set(address, accountCopy)
+    }
+    this.accountStack.push(newTopA)
+    this.codeStack.push(new Map(this.topCodeStack()))
+    this.storageStack.push(new Map(this.topStorageStack()))
+  }
+
+  async getAccount(address: Address): Promise<Account | undefined> {
+    return this.topAccountStack().get(address.toString())
+  }
+
+  async putAccount(address: Address, account?: Account | undefined): Promise<void> {
+    this.topAccountStack().set(address.toString(), account)
+  }
+
+  async deleteAccount(address: Address): Promise<void> {
+    this.topAccountStack().set(address.toString(), undefined)
+  }
+
+  async modifyAccountFields(address: Address, accountFields: AccountFields): Promise<void> {
+    let account = await this.getAccount(address)
+    if (!account) {
+      account = new Account()
+    }
+    account.nonce = accountFields.nonce ?? account.nonce
+    account.balance = accountFields.balance ?? account.balance
+    account.storageRoot = accountFields.storageRoot ?? account.storageRoot
+    account.codeHash = accountFields.codeHash ?? account.codeHash
```

Comment:
I just checked, even if `account` is returned `undefined`, and we provide the `accountFields` which is `Partial<Account>` this still works fine (we default the account to default values)

Reply:
Is there a question here?  I'm not sure what you're getting at?